ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/Node.hs

```diff
@@ -873,14 +928,18 @@ protocolInfoCardano protocolParamsByron@ProtocolParamsByron {
               } = credentials
 
         hotKey <- do
-          let maxKESEvo :: Word64
+          let praosMaxKESEvo = SL.sgMaxKESEvolutions genesisShelley
+
+              maxKESEvo :: Word64
               maxKESEvo = assert (tpraosMaxKESEvo == praosMaxKESEvo) praosMaxKESEvo
 
               startPeriod :: Absolute.KESPeriod
               startPeriod = Absolute.ocertKESPeriod $ praosCanBeLeaderOpCert canBeLeader
 
           HotKey.mkHotKey @m @c initSignKey startPeriod maxKESEvo
```

Comment:
I'm guessing this is the link to the KES-related assertion failure. `mkHotKey` includes a method that calls `signedKES`.

https://github.com/input-output-hk/ouroboros-network/blob/a7f023edadbccf37d980e061c29fb49079bf439e/ouroboros-consensus-protocol/src/Ouroboros/Consensus/Protocol/Ledger/HotKey.hs#L190